### ISSUE TITLE
Start/stop time crashes app in iOS 12.4

### ISSUE THREAD

**Issue by fatmarker** [toopriddy/mytime#366]
From the time management screen, tapping on start/stop crashes the app. The buttons still work though. 

Reproduce error: 
1. Install iOS 12.4
2. Open MyTime Hours screen
3. Tap Start Time
4. (App crashes)
5. Reopen app and go to Hours screen
6. Time was successfully started

**Comment by toopriddy** [comment]
Hello Mark, thank you for the bug report.  Please email me your email
backup by going to MyTime->More->Settings->Email Backup->Email MyTime
Backup to toopriddy@gmail.com
Please do not post it here or reply to this email as your data will be
posted to the internet, please only email it directly to me

I am unable to reproduce your findings and I am wondering if it is related
to your data.  Please also send me what device model you are using (like
iPhone 6S) and if your device is jailbroken or not.

**Comment by fatmarker** [comment]
Backup sent. 
I'm using an iPhone 6.

Thanks for looking into it. Never had the issue before, and nothing changed except updating the iOS from 12.1.2 to 12.4. I did have to restore the phone from an iCloud backup however, so MyTime was reinstalled.

**Comment by toopriddy** [comment]
Received the backup, sadly I am unable to reproduce on an iPhone 6s and will try to find a 6 to try it out on. Here are some additional questions:
1. What language do you have your phone set to? And what region as well so that I know what the time format is
2. Do you have free space on your device?

**Comment by Appaljack** [comment]
Might try uninstalling and reinstalling MyTime. I had to do that recently when I restored an iPad 12.7 from the former iPad 10.5. Actually it was several of the apps that borked. So I had to test and reset quite a few besides.


Gary W. Slocum
423-948-2718 call/text

On Sunday, August 25, 2019, 8:55 PM, toopriddy <notifications@github.com> wrote:

Received the backup, sadly I am unable to reproduce on an iPhone 6s and will try to find a 6 to try it out on. Here are some additional questions:
1. What language do you have your phone set to? And what region as well so that I know what the time format is
2. Do you have free space on your device?

**Comment by fatmarker** [comment]
I’m using about 60% of my storage. 
iOS language is English and the region is Germany.

I restarted the phone and it’s working now. But a related bug has returned that I was experience before the update/restore.

When starting time, I can’t stop it unless I first go to a different view. The same with stopping the time and immediately trying to start it again: first need to switch to another view and return.

**Comment by toopriddy** [comment]
Interesting. The start stop where you have to tap on slightly above the “stop” word to get it to stop is a bug that apple has not yet prioritized to fix. It is a race condition in their library, I was told to work around it and I have yet to find something that works well. 

Another thing you can do is rotate sideways and back and it will be fixed. Sadly apple is in no hurry to fix it :(

**Comment by fatmarker** [comment]
Alright, well thanks for the tip and the quick responses.

**Comment by Appaljack** [comment]
Hi Brent: Would it be fairly easy to add a texting option to the media square dropdown?


Sent from Yahoo Mail for iPhone

**Comment by toopriddy** [comment]
I am sorry I might not be understanding. Are you asking about the share button where there is a box with an arrow jumping out of it?

If so then I would need to look it up. There is some share sheet integration that would have to be done in a handful of places in the app. I can look into it

**Comment by Appaljack** [comment]
Yes, the little box. On many apps it generates a list of things like Copy, Send to Files, iMessage, etc. The reason I thought about it was because Mom might have 300 emails (99% spam) so for her to find mine can be hard but in iMessage, no problem. Be good to send one’s time in as well as I get at times  a call from SGO where he’ll say, ‘cannot find your email’ lol.


Sent from Yahoo Mail for iPhone


On Saturday, September 28, 2019, 8:33 PM, toopriddy <notifications@github.com> wrote:

I am sorry I might not be understanding. Are you asking about the share button where there is a box with an arrow jumping out of it?

If so then I would need to look it up. There is some share sheet integration that would have to be done in a handful of places in the app. I can look into it

**Comment by Appaljack** [comment]
Also on iOS 13.1.1 it was having trouble getting iMail to generate but finally got it going as I found the mail stuck in Drafts. But iMessage gives me a read receipt.


Sent from Yahoo Mail for iPhone


On Saturday, September 28, 2019, 8:33 PM, toopriddy <notifications@github.com> wrote:

I am sorry I might not be understanding. Are you asking about the share button where there is a box with an arrow jumping out of it?

If so then I would need to look it up. There is some share sheet integration that would have to be done in a handful of places in the app. I can look into it

**Comment by podycust** [comment]
its the same on fresh install of the app on iPhone SE running 13.1.1
you get this error in the console app on my Mac 
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[WCSession updateApplicationContext:error:]: applicationContext cannot be nil.'
*** First throw call stack:
(0x1863d498c 0x1860fd0a4 0x1862ca054 0x1a5533878 0x100d9f9fc 0x18a3e3a44 0x189ac8ddc 0x189ac8c68 0x18a3e3a44 0x189e276d0 0x189e27a34 0x189e26a50 0x18a41cd68 0x18a41e0a8 0x18a3faae8 0x18a47223c 0x18a474798 0x18a46d60c 0x1863527e0 0x186352738 0x186351ed0 0x18634d01c 0x18634c8bc 0x1901b8328 0x18a3e26d4 0x100d87278 0x1861d7460)

**Comment by toopriddy** [comment]
@podycust do you have an apple watch paired with your device? was this the first execution of Mytime?

**Comment by podycust** [comment]
no I don't have an apple watch paired and yes it was the first execution of mytime

**Comment by podycust** [comment]
> no I don't have an apple watch paired and yes it was the first execution of mytime

@toopriddy just tried on my main iPhone which I have an Apple Watch paired it doesn’t crash 😀

**Comment by toopriddy** [comment]
there was an update to MyTime 3.11.0 which fixes this and is the first in a long time because of lots of reasons. I hope that now that I have an AI buddy helping me that things will be a lot more stable
